### PR TITLE
Feature/file watching for fun and profit

### PR DESCRIPTION
**Pull request by codejnki**
Code refactor

- Moved the registry management out of main and into the cluster class
  - This may undergo further refactor
- Refactored variable naming
  - Names align more with the concepts of current vs desired
- Commented out statefile for now, not doing anything with it
- Implemented a basic file watcher
  - File watcher has no functionality at the moment
- Added more documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,70 @@ This is a work in progress project which can be used to coordinate deploying the
...
+
+### Arguments
+
+`--loglevel LEVEL` (optional)
+
+This may be set to DEBUG, INFORMATION, WARNING, CRITICAL.  The default level is INFORMATION.
+
+`--configfile /path/to/file.yaml` (required)
+
+The location on disk that nifi-cluster-coordinator to read and process.
```
Comment (frecks) [inline README.md]:
```suggestion
The location on disk of the config file that you want nifi-cluster-coordinator to read and process.
```

File: README.md [hunk 2]
```diff
@@ -4,7 +4,70 @@ This is a work in progress project which can be used to coordinate deploying the
...
+
+### Configuration
+
+#### Clusters
+
+The `clusters` section of the configuration file describes each Apache NiFi instance the cluster coordinator manages.  These instances can be both clusters or stand alone.  For the purposes of the nifi-cluster-coordinator we call them all clusters.
+
+Currently the nifi-cluster-coordinator supports unsecured clusters and clusters secured via certificates.
+
+For secured instances the certificates required to connect to the cluster needs to be accessible by the nifi-cluster-coordinator.
```
Comment (frecks) [inline README.md]:
```suggestion
For secured instances, the certificates required to connect to the cluster need to be accessible by the nifi-cluster-coordinator.
```

File: README.md [hunk 3]
```diff
@@ -4,7 +4,70 @@ This is a work in progress project which can be used to coordinate deploying the
...
+        ssl_key_file: '/foo/foo-root-ca-key.pem'
+  - name: foo-2-cluser
+    host_name: 'http://foo-2-cluster/nifi-api'
+    security:
+      use_certificate: false
+```
+
+#### Apache NiFi Registry
+
+The nifi-cluster-coordinator will insure that the same Apache Nifi Registry entries are present on all managed clusters.
```
Comment (frecks) [inline README.md]:
```suggestion
The nifi-cluster-coordinator will ensure that the same Apache Nifi Registry entries are present on all managed clusters.
```

File: README.md [hunk 4]
```diff
@@ -15,7 +78,7 @@ source .venv/bin/activate
 
 Run `make dev-setup`.
 
-## Configuration
+### Configuration
 
 For local development create a copy of the `nifi-cluster-coordinator.example.yaml` and name it `nifi-cluster-coordinator.yaml` and update it with your cluster configurations.  The `MAKEFILE` is configured out of the box to use a configuration file in that location.
```
Comment (frecks) [inline README.md]:
```suggestion
For local development create a copy of the `nifi-cluster-coordinator.example.yaml` and name it `nifi-cluster-coordinator.yaml`. Update it with your cluster configurations.  The `MAKEFILE` is configured out of the box to use a configuration file in that location.
```

**Review by frecks — COMMENTED** [review]
The code looks generally good for now. I suggested a few small grammatical changes in the README.

We are pretty inconsistent about using single-quoted strings & double-quoted strings in our code. The [PEP 8 Style Guide section on String Quotes](https://www.python.org/dev/peps/pep-0008/#string-quotes) suggests that we should pick one format & stick to it, although they are treated the same in Python. We should pick one now & configure flake8 to warn about usage of the other if that is possible. Do you have a preference between single-quoted strings & double-quoted strings?

File: README.md (hunk 2 above)
Comment (jpreese) [reply README.md]: You guys are awesome.

**Comment by codejnki** [comment]
Agreed.  I really don't have an overwhelming preference. I will take a look at the code and pick one. 

There are some flake8 plugins to help enforce consistent quoting. I will get one of those working.

File: README.md (hunk 2 above)
Comment (frecks) [reply README.md]: Thanks @jpreese, you are too!

**Comment by frecks** [comment]
@codejnki Awesome. I'd like to see you enjoying your day off. I'll get [flake8-quotes](https://pypi.org/project/flake8-quotes/) implemented. If you have no strong preference, I'll suggest that we standardize on single quotes simply because it is the default for flake8-quotes.

**Comment by codejnki** [comment]
That reminds me I forgot to freeze requirements. Need to do that too.

**Comment by frecks** [comment]
@codejnki I added flake8-quotes & fixed all the linter warnings. Giving the README.md suggestions I made a look & updating requirements should be all that's left to do in this PR.

**Comment by codejnki** [comment]
Ok @frecks we are all set on this branch.

**Review by frecks — APPROVED** [review]
All looks good to me.

**Pull request merged**